### PR TITLE
Partner-related things

### PR DESCRIPTION
**Pull request by nanaian**

### CODE REVIEW

File: include/variables.h [hunk 1]
```diff
@@ -12,7 +12,7 @@ extern PlayerData gPlayerData;
 extern ActionState gPlayerActionState;
 extern PlayerAnim gPlayerAnimation;
 extern PlayerStatus gPlayerStatus;
-extern PlayerStatus* gPlayerStatusPtr;
+PlayerStatus* gPlayerStatusPtr;
```
Comment (ethteck) [inline include/variables.h:15]: whuh

File: src/world/partners.c [hunk 2]
```diff
@@ -214,3 +244,159 @@ WorldPartner gWorldPartners[] = {
         .canPlayerPause = world_partner_can_use_ability_default,
     },
 };
+
+
+NpcId create_basic_npc(NpcBlueprint* blueprint);
+
+INCLUDE_ASM(s32, "code_838b0_len_5900", use_consumable);
```
Comment (ethteck) [inline src/world/partners.c:251]: code_838b0_len_5900 these refs should be renamed and the asm moved

**Review by ethteck — DISMISSED** [review]

File: src/world/actions.c [hunk 3]
```diff
@@ -1,3 +1,82 @@
...
+s32 D_800F7B4C = 0;
+
+s32 D_800F7B50[] = { 0x417C1B33, 0xC0EC5C14, 0x405C9AAA, 0xBF400000 };
+s32 D_800F7B60[] = { 0x3E1E0C17, 0xBEB33DB0, 0xBE3AA2E4, 0x3C3CBE62 };
+s32 D_800F7B70[] = { 0x40000000, 0x40800000, 0x42000000, 0xC2000000 };
+s32 D_800F7B80[] = { 0x00000000, 0x00000000, 0x00000000, 0x00000000 }; // possibly s16
+s32 D_800F7B90[] = { 0x00000000, 0x00390000, 0x00390002, 0x00390003, 0x00390004, 0x00390005, 0x00390008, 0x00390007, 0x00390013, 0x00390014, 0xFFFFFFFF, 0x005A0000, 0x005A0002, 0x005A0003, 0x005A0004, 0x005A0006, 0x005A0007, 0x005A0009, 0x005A000A, 0x005A000B, 0x005A001A, 0xFFFFFFFF, 0x00580000, 0x00580001, 0x00580002, 0x00580004, 0x00580005, 0x00580006, 0x00580007, 0x00580008, 0x00580009, 0x00580012, 0x00580014, 0x0058001B, 0xFFFFFFFF };
+s32 D_800F7C1C[] = { 0x00580001, 0x00580004, 0x00580006, 0x00580008, 0x00580000 };
+s32 D_800F7C30[] = { 0x0058001B, 0x00580001, 0x00580004, 0x00580006, 0x00580008, 0x00580000, 0x0058001B, 0x005A0003, 0x005A0006, 0x005A0009, 0x005A000B, 0x005A0000, 0x005A001A, 0x00390002, 0x00390003, 0x00390004, 0x00390005, 0x00390000, 0x00390014 };
+s32 D_800F7C7C[] = { 0x800F7BE8, 0x800F7BE8, 0x800F7BBC, 0x800F7B94 };
```
Comment (pixel-stuck) [inline src/world/actions.c:34]: pointers here.

File: src/world/actions.c [hunk 4]
```diff
@@ -1,3 +1,82 @@
...
+s32 func_802B6478_E28EA8();
+s32 func_802B60A4_E29514();
+s32 func_802B609C_E28ACC();
+
+f32 D_800F7B40 = 0.0f;
+s32 D_800F7B44 = 0;
+f32 D_800F7B48 = 0.0f;
+s32 D_800F7B4C = 0;
+
+s32 D_800F7B50[] = { 0x417C1B33, 0xC0EC5C14, 0x405C9AAA, 0xBF400000 };
```
Comment (pixel-stuck) [inline src/world/actions.c:27]: these look like floats

File: src/world/actions.c [hunk 5]
```diff
@@ -1,3 +1,82 @@
...
+s32 D_800F7B60[] = { 0x3E1E0C17, 0xBEB33DB0, 0xBE3AA2E4, 0x3C3CBE62 };
```
Comment (pixel-stuck) [inline src/world/actions.c:28]: potentially also floats... not 100% sure tho

File: src/world/actions.c [hunk 6]
```diff
@@ -1,3 +1,82 @@
...
+s32 D_800F7B70[] = { 0x40000000, 0x40800000, 0x42000000, 0xC2000000 };
```
Comment (pixel-stuck) [inline src/world/actions.c:29]: more potential floats

File: src/world/partner/goombario.c [hunk 7]
```diff
@@ -61,7 +61,7 @@ ApiStatus func_802BDB84(ScriptInstance* script, s32 isInitialCall) {
 
 // Something is up with D_8010EBB0 I think. It might be a struct or something
 #ifdef NON_MATCHING
-void world_goombario_pre_battle(s32 arg0) {
+void world_goombario_pre_battle(Npc* partner) {
```
Comment (pixel-stuck) [inline src/world/partner/goombario.c:64]: arg0 isn't renamed in the body of the func.

File: src/world/partners.c [hunk 8]
```diff
@@ -32,10 +33,39 @@
 s32 world_partner_can_use_ability_default(Npc* partner);
 s32 world_partner_can_player_pause_default(Npc* partner);
 
-static s32 _pad[] = { 0x00, 0x00, 0x00 };
+// Partner icons
+s32 D_800F7F00[] = {
+    0x80107CA8, 0x80107CF8, 0x80107D48, 0x80107D98, 0x80107DE8, 0x80107CA8, 0x80107E88, 0x80107ED8, 0x80107F28, 0x80107E38, 0x80107FC8, 0x80107FC8, 0x80107FC8, 0x80107FC8, 0x80107FC8, 0x80107FC8,
```
Comment (pixel-stuck) [inline src/world/partners.c:38]: pointers, here and below

File: src/world/partners.c [hunk 9]
```diff
@@ -32,10 +33,39 @@
...
-WorldPartner gWorldPartners[] = {
-    // XXX: it's possible that's there's a "none" entry here to match up with enum PartnerID
+s32 D_800F7FE8 = -1;
+s32 D_800F7FEC = 1;
+s32 D_800F7FF0 = 2;
+s32 D_800F7FF4 = 4;
+s32 D_800F7FF8 = 5;
+s32 D_800F7FFC = 7;
+s32 D_800F8000[] = { 8, 0, 0, 0 };
+s32 D_800F8010[] = { 0x003251D0, 0x00325AD0, 0x802C05CC, 0x00000000 };
```
Comment (pixel-stuck) [inline src/world/partners.c:58]: pointer

**Review by pixel-stuck — CHANGES_REQUESTED** [review]

**Pull request merged**